index.js

```diff
@@ -113,4 +117,76 @@ app.get('/squadStats', async (req, res) => {
 
     res.send(retVal);
 
-});
\ No newline at end of file
+});
+
+//stats squad individual
+app.get('/squadStatsIndiv', async (req, res) => {
+
+    let gt = req.query.gt;
+
+    let result = await lib.halo.mcc['@0.0.11'].squad.activity({
+        gamertags: [
+            'sagwasswagga',
+            'Cragle 3',
+            'A Pregnant Nunn',
+            'babaganoosh069',
+            'Girthy Jello',
+            'SirPuffingtin',
+            'ya boi nar',
+            'l swordzzz'
+        ],
+        minSquadSize: '3',
+        game: 'Halo 3'
+    });
+
+
+    let games = result.games;
+
+    let aggregSqdStats = [];
+    
+    for(let match of games) {
+        let win = match.squadWon;
+        let avgHeadshotRate = 0.0;
+        let avgKD = 0.0;
+        let avgScore = 0.0;
+        let avgKills = 0.0;
+        let avgDeaths = 0.0;
+        let avgAssists = 0.0;
+        let playedAt = match.playedAt;
```

Comment:
should avgs be outside this loop? also you never divide avgs by number of matches

Reply:
no the averages need to be inside this loop, because the mcc api returns an array of games, and each game has an array of stats for the requested gamertags. Therefore, in order to get the aggregate across the squad, we need to loop over all the gamertags we queried (say you got 2.3 KD and I got 1.3 KD, we need to average those and get 1.8 KD)